modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -206,11 +214,11 @@ extension AddressBookClient: DependencyKey {
         return syncedContacts
```

Comment:
On line 211, it's only necessary to store the contacts locally if the merge added anything relative to the original contents of the local file. More importantly (because remote storage has higher latency), it's only necessary to store the contacts remotely if the merge added anything relative to the original contents of the remote file.

It might sound as though this optimization makes things more complicated. But actually it makes them simpler, because then you no longer need the `remoteStore` flag. In most cases, the remote storage will be synced with the local storage already, and we won't do a remote store. If it happens not to be synced, then this approach (where we remove the `remoteStore` flag) means that in the case of a mutation, we might do two remote stores — one to merge in the local contacts and one to make the update. But that case should be so rare that it doesn't matter for efficiency.